src/python/module/smartredis/util.py

```diff
@@ -111,3 +114,136 @@ def typecheck(arg, name, _type):
     """
     if not isinstance(arg, _type):
         raise TypeError(f"Argument {name} is of type {type(arg)}, not {_type}")
+
+
+
+class DatasetConverter:   
+    
+    def add_metadata_for_xarray(ds,data_names,dim_names,coord_names=None,attr_names=None):
+        """Extracts metadata from a SmartRedis dataset and adds it to
+        dataformat specific fieldnames 
+
+        :param ds: a SmartRedis Dataset with metadata added to Xarray 
+        specific field names 
+        :type ds: SmartRedis Dataset
+        :param data_names: variable data field name
+        :type data_names: list or string 
+        :param dim_names: dimension field names. Defaults to None.
+        :type dim_names: list
+        :param coord_names: coordinate field names. Defaults to None.
+        :type coord_names: list, optional
+        :param attr_names: attribute field names. Defaults to None.
+        :type attr_names: list, optional
+        """        
+        
+        args = ['dim_names','coord_names','attr_names'] 
+
+        for d in data_names:  
+
+            ds.add_meta_string("_xarray_data_names",d)
+   
```

Comment:
You can remove some of these empty lines, particularly the ones right after the `for` loops